maintenance-editor/src/main/java/ca/on/oicr/gsi/shesmu/maintenance/Editor.java

```diff
@@ -0,0 +1,196 @@
+      return;
+    }
+    new Editor(file).setVisible(true);
+  }
+
+  public Editor(File file) throws IOException {
+    setTitle(file.toString() + " - Maintenance Schedule Editor");
+    setSize(700, 400);
+    setDefaultCloseOperation(JFrame.DISPOSE_ON_CLOSE);
+    final JPanel panel = new JPanel(new BorderLayout());
+    add(panel);
+    final JToolBar toolbar = new JToolBar();
+    panel.add(toolbar, BorderLayout.PAGE_START);
+    final JButton add = new JButton("Add");
+    toolbar.add(add);
+    final JButton delete = new JButton("Delete");
+    toolbar.add(delete);
+    final JButton reload = new JButton("Revert");
+    toolbar.add(reload);
+    final JButton save = new JButton(UIManager.getIcon("FileView.floppyDriveIcon"));
+    toolbar.add(save);
+    save.setToolTipText("Save");
+    final DateTableModel model = new DateTableModel();
+    final JTable table =
+        new JTable(model) {
+          private static final long serialVersionUID = 1L;
+
+          public Component prepareRenderer(TableCellRenderer renderer, int row, int column) {
+            final JComponent c = (JComponent) super.prepareRenderer(renderer, row, column);
+            if (model.rows.get(row)[0].isAfter(model.rows.get(row)[1])) {
+              c.setBorder(BorderFactory.createMatteBorder(1, 1, 1, 1, Color.RED));
+            }
+            return c;
+          }
+        };
+    final DateTimeTableEditor editor = new DateTimeTableEditor();
+    editor.clickCountToEdit = 0;
+    table.setDefaultEditor(LocalDateTime.class, editor);
+    table.setDefaultRenderer(LocalDateTime.class, new DateTimeTableEditor());
+    table.setFillsViewportHeight(true);
+    panel.add(new JScrollPane(table), BorderLayout.CENTER);
+    final Runnable loadFile =
+        () -> {
+          try (Stream<String> lines = Files.lines(file.toPath())) {
+            model.rows.clear();
+            lines //
+                .map(TAB::split) //
+                .filter(x -> x.length > 1) //
+                .forEach(x -> model.rows.add(new LocalDateTime[] {parse(x[0]), parse(x[1])}));
+          } catch (IOException e) {
+            JOptionPane.showMessageDialog(
+                this,
+                e.getMessage(),
+                "Save Failed",
+                JOptionPane.ERROR_MESSAGE,
+                UIManager.getIcon("OptionPane.errorIcon"));
+          }
+        };
+    loadFile.run();
+
+    add.addActionListener(
+        event -> {
+          model.rows.add(new LocalDateTime[] {LocalDateTime.now(), LocalDateTime.now()});
+          model.fireTableDataChanged();
+        });
+    delete.addActionListener(
+        event -> {
+          Arrays.stream(table.getSelectedRows()).forEach(model.rows::remove);
+          model.fireTableDataChanged();
+        });
+    save.addActionListener(
+        event -> {
+          try (PrintWriter writer = new PrintWriter(file)) {
+            for (LocalDateTime[] row : model.rows) {
+              writer.write(toUtc(row[0]));
```

Comment:
From what I've read about LocalDateTime, I'm not sure it takes DST into account. So are time changes still up to the user to configure?

Reply:
It should do timezone correction. I read the file in `ZonedDateTime`, which does the DST correction when I go to an from `LocalDateTime`.